pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/SierraContributors.scala

```diff
@@ -37,82 +46,92 @@ object SierraContributors
-            }
-        } else {
-          getPerson(subfields, normalisePerson = true)
-            .map { person =>
-              identify(subfields, person, "Person")
-            }
-        }
+      val maybeAgent = if (hasSubfieldT) {
+        getLabel(subfields)
+          .map {
+            Agent(_)
+          }
+          .map { agent =>
+            identify(subfields, agent, "Agent")
+          }
+      } else {
+        getPerson(subfields, normalisePerson = true)
+          .map { person =>
+            identify(subfields, person, "Person")
+          }
+      }
 
-        maybeAgent map { agent =>
-          Contributor(
-            agent = agent,
-            roles = roles
-          )
-        }
+      maybeAgent map { agent =>
+        Contributor(
+          agent = agent,
+          roles = roles
+        )
       }
-  }
+    }
 
   /* For a given MARC tag (110 or 710), return a list of all the Contributor[Organisation] instances
    * this MARC tag represents.
    */
   private def getOrganisationContributors(
     bibData: SierraBibData,
-    marcTag: String): List[Contributor[MaybeDisplayable[Organisation]]] = {
-    val organisations = getMatchingSubfieldLists(
+    marcTag: String): List[Contributor[MaybeDisplayable[Organisation]]] =
+    getMatchingSubfieldLists(
       bibData,
       marcTag = marcTag,
       marcSubfieldTags = List("a", "b", "c", "d", "e", "0")
-    )
+    ).flatMap { subfields: List[MarcSubfield] =>
+      val roles = getContributionRoles(subfields)
+      val maybeAgent = getOrganisation(subfields)
+
+      maybeAgent.map { agent =>
+        Contributor(
+          agent = identify(subfields, agent, "Organisation"),
+          roles = roles
+        )
+      }
+    }
 
-    organisations
-      .flatMap { subfields: List[MarcSubfield] =>
-        val roles = getContributionRoles(subfields)
-        val maybeAgent = getOrganisation(subfields)
+  private def getMeetingContributors(
+    bibData: SierraBibData,
+    marcTag: String): List[Contributor[MaybeDisplayable[Meeting]]] = {
+    getMatchingSubfieldLists(
+      bibData,
+      marcTag = marcTag,
+      marcSubfieldTags = List("a", "c", "d", "j", "t", "0")
+    ).flatMap { subfields: List[MarcSubfield] =>
+      val roles = getContributionRoles(subfields, "j")
+      val maybeAgent = getMeeting(subfields)
 
-        maybeAgent.map { agent =>
-          Contributor(
-            agent = identify(subfields, agent, "Organisation"),
-            roles = roles
-          )
-        }
+      maybeAgent.map { agent =>
+        Contributor(
+          agent = identify(subfields, agent, "Meeting"),
+          roles = roles
+        )
       }
+    }
   }
 
   private def getContributionRoles(
-    subfields: List[MarcSubfield]): List[ContributionRole] = {
-    // Extract the roles from subfield $e.  This is a repeatable field.
+    subfields: List[MarcSubfield],
+    subfieldTag: String = "e"): List[ContributionRole] =
```

Comment:
Is this a logical default, or just to help with backwards compatibility?

Reply:
This was uses by the previous two (`organisation` and `person`). I could remove the default and make it explicit if you think makes more sense?

Reply:
Having these funny floaty letters always feels odd, I would be explicit as there's nothing in the spec or elsewhere saying there is any consistency with Marc...